src/core/stream.c

```diff
@@ -838,6 +898,39 @@ QuicStreamParamGet(
         Status = QUIC_STATUS_SUCCESS;
         break;
     }
+
+    case QUIC_PARAM_STREAM_RELIABLE_OFFSET:
+        if (*BufferLength < sizeof(uint64_t)) {
+            *BufferLength = sizeof(uint64_t);
+            Status = QUIC_STATUS_BUFFER_TOO_SMALL;
+            break;
+        }
+        if (Buffer == NULL) {
+            Status = QUIC_STATUS_INVALID_PARAMETER;
+            break;
+        }
+        *(uint64_t*) Buffer = Stream->ReliableOffsetSend;
```

Comment:
Should we return "invalid state" error for this and recv offset if they haven't been set? While zero is understood internally to mean "not set" I think externally it should be treated as an invalid value.

Reply:
Correct if I am wrong, we agreed adding the GetParam for (send / receive) here is for internal testing purposes only (Added specifically for logging). Externally, apps don't really care about these internal details. Also, since I added GetParam tests in ApiTest, adding this requirement would complicate the logic a lot.

Reply:
This is a public, not private interface. While it's mainly for set, it is available for query, so we have to decide what we want the official behavior to be. What should the result of querying this offset be if the app never set it? Does it make sense to indicate `0` to the app?

Reply:
Invalid if not set sounds good